node/src/frost_sign.rs

```diff
@@ -0,0 +1,576 @@
+    impl Protocol for NonActiveParticipant {
+        type Output = FrostOutput;
+        fn poke(&mut self) -> Result<Action<Self::Output>, ProtocolError> {
+            Ok(Action::Wait)
+        }
+        fn message(&mut self, _: Participant, _: MessageData) {}
+    }
+
+    fn build_protocols(
+        max_signers: usize,
+        min_signers: usize,
+        non_active_participant: usize,
+        coordinators: usize,
+    ) -> Vec<(Participant, Box<dyn Protocol<Output = FrostOutput>>)> {
+        let active_participants = max_signers - non_active_participant;
+        assert!(active_participants >= min_signers);
+        let msg = "hello_near";
+        let msg_hash = hash(msg.as_bytes());
+
+        let mut identifiers = Vec::with_capacity(max_signers);
+        for i in 1..max_signers + 1 {
+            // from 1 to avoid assigning 0 to a ParticipantId
+            identifiers.push(ParticipantId::from_raw((10 * i) as u32))
+        }
+
+        let frost_identifiers = identifiers
+            .iter()
+            .map(|&x| to_frost_identifier(x.into()))
+            .collect::<Vec<_>>();
+
+        let mut rng = thread_rng();
+        let (shares, pubkey_package) = frost_ed25519::keys::generate_with_dealer(
+            max_signers as u16,
+            min_signers as u16,
+            frost_ed25519::keys::IdentifierList::Custom(&frost_identifiers),
+            &mut rng,
+        )
+        .unwrap();
+
+        let key_packages = shares
+            .iter()
+            .map(|(id, share)| {
+                (
+                    id,
+                    frost_ed25519::keys::KeyPackage::try_from(share.clone()).unwrap(),
+                )
+            })
+            .collect::<BTreeMap<_, _>>();
+
+        let mut protocols: Vec<(Participant, Box<dyn Protocol<Output = FrostOutput>>)> =
+            Vec::with_capacity(max_signers);
+
+        for i in 0..non_active_participant {
+            protocols.push((identifiers[i].into(), Box::new(NonActiveParticipant {})))
+        }
+
+        for i in non_active_participant..max_signers {
+            protocols.push((
+                identifiers[i].into(),
+                frost_sign(
+                    rng.clone(),
+                    i >= max_signers - coordinators,
+                    identifiers[i],
+                    key_packages[&frost_identifiers[i]].clone(),
+                    pubkey_package.clone(),
+                    min_signers,
+                    msg_hash.as_bytes().to_vec(),
+                ),
+            ))
+        }
+
+        protocols
+    }
+
+    #[test]
+    fn basic_two_participants() {
+        let max_signers = 2;
+        let min_signers = 2;
+        let non_active_participants = 0;
+        let coordinators = 1;
+
+        let protocols = build_protocols(
+            max_signers,
+            min_signers,
+            non_active_participants,
+            coordinators,
+        );
+        run_protocol(protocols).unwrap();
```

Comment:
Is it possible to sanity check the returned results? (One is a coordinator return result, and the others are participants or something like that?)